### PR TITLE
PhysicsFS 3.0.0

### PR DESCRIPTION
**Pull request by Paril**
- Minizip now uses miniz just to keep things all-in-one and not require zlib. Makes it easy for Winders. If this isn't what jdolan wants I'll modify it to require zlib like before and figure out what I'll do on the Windows side for that. Since it's the only dep we have that uses zlib anyways...
- Used the non-deprecated versions of the functions for PhysFS.
- Enumerate now allows the return values to stop, error or continue enumeration.

### CODE REVIEW

**Comment by kaadmy** [comment]
Clean code, I figured there would be more changes to get Physfs3 working.

File: src/filesystem.h [hunk 1]
```diff
@@ -41,7 +41,13 @@ typedef struct {
...
 } file_t;
 
-typedef void (*Fs_EnumerateFunc)(const char *path, void *data);
+typedef enum {
+	FS_ENUM_ERROR = -1,
+	FS_ENUM_STOP = 0,
+	FS_ENUM_CONTINUE = 1
+} Fs_EnumerateResult;
+
+typedef Fs_EnumerateResult (*Fs_EnumerateFunc)(const char *path, void *data);
```
Comment (jdolan) [inline src/filesystem.h]: I guess this might be useful someday. Might as well expose it.

File: src/filesystem.h [hunk 2]
```diff
@@ -65,7 +71,7 @@ int64_t Fs_Load(const char *filename, void **buffer);
 void Fs_Free(void *buffer);
 _Bool Fs_Rename(const char *source, const char *dest);
 _Bool Fs_Unlink(const char *filename);
-void Fs_Enumerate(const char *pattern, Fs_EnumerateFunc, void *data);
+int32_t Fs_Enumerate(const char *pattern, Fs_EnumerateFunc, void *data);
```
Comment (jdolan) [inline src/filesystem.h]: I feel like the return type here could be either FsEnumerateResult or _Bool.

**Comment by Paril** [comment]
Glib comprises like 400-some source files, and SDL comprises like 6 different modules that we use (and is in C++), so those are reasons why I'd say no on those, but PhysFS is fairly small and can easily be added as a submodule (well, it *could*, if it wasn't Mercurial...) or something? Just thinking out loud.
Doesn't have to be bundled, but would be cool if it did.

**Comment by Paril** [comment]
Cou

**Comment by Paril** [comment]
Whoopsies, wrong button.

File: src/filesystem.h (hunk 2 above)
Comment (Paril) [reply src/filesystem.h]: _Bool is preferable, yeah. I just picked int32_t because that's what PhysFS returns, but it only returns 0 or 1.

**Comment by Paril** [comment]
Now is lib like before! Needs Unix and Linux build path though @jdolan

**Pull request closed without merging**